### PR TITLE
Consolidate AssociativeFlatten instances

### PR DESCRIPTION
**Pull request by mtsokol**
@sideeffffect Hi!

Here's PR for #284, I tried to follow previous work.

### CODE REVIEW

File: src/main/scala/zio/prelude/NonEmptyList.scala [hunk 1]
```diff
@@ -435,16 +435,6 @@ object NonEmptyList extends LowPriorityNonEmptyListImplicits {
   implicit def NonEmptyListAssociative[A]: Associative[NonEmptyList[A]] =
     Associative.make(_ ++ _)
 
-  /**
```
Comment (lemastero) [inline src/main/scala/zio/prelude/NonEmptyList.scala:438]: I would keep `NonEmptyListIdentityFlatten` and `ZPureIdentityFlatten` in their companion objects.

@sideeffffect WDYT?

Comment (adamgfraser) [reply src/main/scala/zio/prelude/NonEmptyList.scala:438]: Yes, we should put instances in the companion object whenever possible.

Comment (adamgfraser) [reply src/main/scala/zio/prelude/NonEmptyList.scala:438]: Yes, we should put instances in the companion object whenever possible.

**Review by sideeffffect — CHANGES_REQUESTED** [review]
`NonEmptyListIdentityFlatten` and `ZPureIdentityFlatten` should stay in `NonEmptyList` and `ZPure` respectively

File: src/main/scala/zio/prelude/NonEmptyList.scala (hunk 1 above)
Comment (mtsokol) [reply src/main/scala/zio/prelude/NonEmptyList.scala]: Done!

**Comment by mtsokol** [comment]
@sideeffffect Sure, done!

**Review by sideeffffect — APPROVED** [review]

**Review by lemastero — APPROVED** [review]

**Comment by sideeffffect** [comment]
@adamgfraser this is ready to be merged. Could you please have a look at it? :pray:

**Review by adamgfraser — APPROVED** [review]
Thank you! :pray:

**Pull request merged**